### PR TITLE
auth websocket and spawnSync implementation

### PR DESCRIPTION
**Pull request by xavluiz**
This request contains updates to use the spawnSync, result safety checks to prevent host editor crash issues, use auth.js for the jwt and email value, and fixed the end of day report check for weekend use cases

### CODE REVIEW

File: src/Util.ts [hunk 1]
```diff
@@ -359,35 +367,35 @@ export function getOs() {
...
+  return execCmd(cmd, opts);
 }
 
-export async function getCommandResultList(cmd, projectDir = null) {
-  let result = await wrapExecPromise(`${cmd}`, projectDir);
-  if (!result) {
-    return [];
+function execCmd(cmd, opts) {
+  try {
+    const data = execSync(cmd, opts);
```
Comment (brettmstevens7) [inline src/Util.ts:384]: Are there any large outputs from commands we're sending through here (like Git?) that would be better suited using a spawn? I'm reading up on this now 🤓

Comment (brettmstevens7) [reply src/Util.ts:384]: Found an example with git log >> https://davidwalsh.name/catching-fatal-errors-nodejs-childprocess

**Review by brettmstevens7 — COMMENTED** [review]
Nice, seems like you've narrowed in on the crux of the performance issues. Curious to hear @mmclead and @pbstriker38's thoughts.

File: src/Util.ts (hunk 1 above)
Comment (xavluiz) [reply src/Util.ts:384]: yeah, i've looked into this and I think I'll change it to use the spawn if that person found that only spawn can truly handle exceptions. the bulk of this work has been to ensure the command executions happen in one place, so we'll make sure that one place is solid. good find, thanks Brett.

Comment (xavluiz) [reply src/Util.ts:384]: looks like all of our command structure will need to change. the command and an array of options for that command.

child_process.spawnSync(command[, args][, options])
i.e.
var ls = cp.spawnSync('ls', ['-l', '/usr'], { cwd: "./", encoding : 'utf8' });

vs

child_process.execSync(command[, options])
var ls = cp.execSync('ls -l /usr', {cwd: "./", encoding : 'utf8'})

Comment (xavluiz) [reply src/Util.ts:384]: I tested it out and it works as well, but we'll have to change that method's logic to slice out the options. Here's the proposed function

function execCmd(cmd = "", opts = {}) {
  let resultStr = "";
  try {
    // add utf 8 to the options
    opts = {
      ...opts,
      encoding: "utf8",
    };

    // get the command and options
    let cmdOpts = cmd.trim().split(" ");
    if (cmdOpts.length) {
      // get the command
      cmd = cmdOpts[0];
      if (cmdOpts.length > 1) {
        // splice out the options
        cmdOpts = cmdOpts.slice(1, cmdOpts.length);
      } else {
        // no options, set the array to empty
        cmdOpts = [];
      }
    }
    const data = spawnSync(cmd, cmdOpts, opts);
    if (data) {
      const errorText = data?.stderr.toString().trim() ?? null;
      if (errorText) {
        console.error("command error: ", errorText);
        return null;
      }
      resultStr = data?.stdout.toString().trim() ?? "";
      const lines = resultStr ? resultStr.split(/\r?\n/) : null;
      if (lines && lines.length) {
        resultStr = lines[0];
      }
    }
  } catch (e) {
    console.error("command error: ", e);
  }
  return resultStr;
}

Comment (xavluiz) [reply src/Util.ts:384]: i'll update the PR, it'll be easier to view

**Pull request merged**